spring-session-data-mongodb/src/main/java/org/springframework/session/data/mongo/ReactiveMongoSessionRepository.java

```diff
@@ -94,17 +100,20 @@ public ReactiveMongoSessionRepository(ReactiveMongoOperations mongoOperations) {
 	@Override
 	public Mono<MongoSession> createSession() {
 
-		return Mono.justOrEmpty(this.defaultMaxInactiveInterval.toSeconds()) //
-				.map(MongoSession::new) //
-				.doOnNext((mongoSession) -> publishEvent(new SessionCreatedEvent(this, mongoSession))) //
-				.switchIfEmpty(Mono.just(new MongoSession()));
+		return Mono.justOrEmpty(this.defaultMaxInactiveInterval.toSeconds())
+				.map((interval) -> new MongoSession(this.sessionIdGenerationStrategy.generate(), interval))
+				.doOnNext((mongoSession) -> publishEvent(new SessionCreatedEvent(this, mongoSession)))
+				.switchIfEmpty(Mono.just(new MongoSession(this.sessionIdGenerationStrategy.generate())));
```

Comment:
I think we do not need a reactive equivalent since now the strategy is inside the session implementation

Reply:
Is blocking caused by /dev/random? If so then fix the root cause, not the symptom.
=> Add an entropy generator in your VM

Caveat: If using Docker, put the generator in the host, and all containers will benefit from that shared kernel entropy pool.

Entropy generators for Linux:
1. Linux kernel 5.4+ => A new JITTER entropy generator was added. Many LTS/EL kernels back-ported that enhancement.
2. Linux kernel before 5.4 => Install haveged (JITTER), or rngd (JITTER, plus optional x86-only RDRAND)

Caveat: If you care about FIPS 140, disable all JITTER and only use a compliant entropy source (e.g. RDRAND on x86, or other third-party compliant entropy daemons/services).
Caveat: I have not seen SecureRandom blocking issues on macOS or Windows for years.

Reply:
RHEL case study:
- RHEL 9.x w/ 5.14 => Built-in since 9.0 because kernel 5.4+ comes with JITTER
- RHEL 8.x w/ 4.18 => Back-ported in a later 8.x, so update to latest 8.x
- RHEL 7.x w/ 3.10 => Back-ported in a later 7.x, so update to latest 7.x
- RHEL 6.x w/ 2.6 => Not back-ported, manually install haveged or rng-tools

Reply:
If you are on an old Linux kernel, and can't install haveged/rng-tools, fall back to this old Java hack:
3. JVM system property at startup:`-Djava.security.egd=file:/dev/urandom`

Caveat: This is a security concern. It will lead to weak random data, especially right after a reboot when the entropy pool is empty.
Caveat: It probably only works for the default Sun/Oracle providers. Third-party crypto providers (e.g. Bouncy Castle) may or may not offer a workaround.